internal/pkg/cli/app_deploy.go

```diff
@@ -248,15 +261,44 @@ func (opts *appDeployOpts) sourceEnvName() error {
 		opts.env = selectedEnvName
 	}
 
-	for _, envName := range envNames {
-		if opts.env == envName {
+	for _, env := range opts.projectEnvironments {
+		if opts.env == env.Name {
+			opts.targetEnvironment = env
+
 			return nil
 		}
 	}
 
 	return fmt.Errorf("invalid env name: %s", opts.env)
 }
 
+func (opts *appDeployOpts) configureClients() error {
+	// ECR client against tools account profile AND target environment region
+	ecrSess, err := session.DefaultWithRegion(opts.targetEnvironment.Region)
+	if err != nil {
+		return fmt.Errorf("create ECR session with region %s: %w", opts.targetEnvironment.Region, err)
+	}
+	opts.ecrService = ecr.New(ecrSess)
+
+	// CF client with target environment assume role AND target environmnet region
+	// TODO: expose the EnvManagerRole
+	// appDeployCfSess, err := session.FromRole(opts.targetEnvironment.EnvManagerRole, opts.targetEnvironment.Region)
```

Comment:
oh cool, so we need to store the roleARN in SSM right?

Reply:
Right, I'll create an issue for that if there isn't one already. Ideally this "app deploy CF" client will be configured with an assumed role session and the target environment region.

Reply:
Actually that may no longer be the case since ECR repo creation has moved to StackSets and the access scope looks like it's the tools account, so I can go ahead and update this to use a default session against the target environment region.